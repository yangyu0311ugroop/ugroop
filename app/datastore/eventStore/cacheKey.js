const filterFlightBookingsByTemplateId = ({ idsProp, templateIdProp }) => ({
  [idsProp]: ids,
  [templateIdProp]: templateId,
}) =>
  `event.events.flightBookings.${
    ids ? ids.toString() : null
  }.${templateId}.filterFlightBookingsByTemplateId`;

const filterEventsByFlightBookingDataId = ({
  eventStartTimesProp,
  eventFlightBookingDataIdsProp,
  flightBookingDataIdProp,
  allFlightsProp,
}) => ({
  [eventStartTimesProp]: eventStartTimes,
  [eventFlightBookingDataIdsProp]: eventDataIds,
  [flightBookingDataIdProp]: flightBookingDataId,
  [allFlightsProp]: allFlights,
}) =>
  `event.events.flightBookings.${
    eventStartTimes ? eventStartTimes.toString() : null
  }.${eventDataIds ? eventDataIds.toString() : null}.${
    flightBookingDataId ? flightBookingDataId.toString() : null
  }.${allFlights}.filterEventsByFlightBookingDataId`;

export const EVENT_STORE_CACHE_KEYS = {
  filterFlightBookingsByTemplateId,
  filterEventsByFlightBookingDataId,
};
