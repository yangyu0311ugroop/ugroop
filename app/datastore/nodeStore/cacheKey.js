import { ARRAY_HELPERS } from 'utils/helpers/arrays';
import { SELECTOR_HELPERS } from 'utils/helpers/selectors';

const remainingCheckitemsCount = ({ checklists, statuses, type }) => {
  if (!Array.isArray(checklists) || !Array.isArray(statuses)) {
    return undefined;
  }

  return `nodes.${checklists.toString()}.${statuses.toString()}.${type ||
    ''}.outstanding_completed.count`;
};
const openClosedCount = ({ id }) =>
  Array.isArray(id) ? `nodes.${id.toString()}.statuses.open_closed` : '';

const checklists = ({ id }) => `node.${id}.checklists`;

const allChecklists = ({ id }) => {
  if (!id) return undefined;

  if (Array.isArray(id)) {
    return `nodes.${id.toString()}.checklists`;
  }
  return allChecklists({ id: [id] });
};

const allStatuses = ({ id }) => {
  if (!id) return undefined;

  if (Array.isArray(id)) {
    return `nodes.${id.toString()}.statuses`;
  }
  return allStatuses({ id: [id] });
};

const cachedChildren = ({ idProp }) => ({ [idProp]: id }) =>
  `node.${id}.cachedChildren`;

const cachedIndex = ({ idProp, parentNodeIdProp }) => ({
  [idProp]: id,
  [parentNodeIdProp]: parentNodeId,
}) => `node.${parentNodeIdProp}:${parentNodeId}.${idProp}:${id}.index`;

const eventDataIds = ({ idsProp }) => ({ [idsProp]: ids }) =>
  `node.${idsProp}:${ids ? ids.toString() : 'null'}.eventDataIds`;

const addEventDataIds = ({ nodesProp }) => ({ [nodesProp]: nodes }) =>
  `node.${nodesProp}:${
    nodes ? nodes.map(({ id }) => id).toString() : 'null'
  }.addEventDataIds`;

const filterByTypes = ({ ids, types }) => ({
  [ids]: idsPropValue,
  [types]: typesPropValue,
}) => {
  const idsValue = SELECTOR_HELPERS.propOrValueArray(ids, idsPropValue);
  const typesValue = SELECTOR_HELPERS.propOrValueArray(types, typesPropValue);
  return `node.${ids}:${idsValue.toString()}.${types}:${typesValue.toString()}.filterByTypes`;
};

const filterByStatuses = ({ ids, statuses }) => ({
  [ids]: idsPropValue,
  [statuses]: statusesPropValue,
}) => {
  const idsValue = SELECTOR_HELPERS.propOrValueArray(ids, idsPropValue);
  const statusesValue = SELECTOR_HELPERS.propOrValueArray(
    statuses,
    statusesPropValue,
  );
  return `node.${ids}:${idsValue.toString()}.${statuses}:${statusesValue.toString()}.filterByStatuses`;
};

const filterAndSortByTime = ({ idProp, dateProp, nodesProp }) => ({
  [idProp]: id,
  [dateProp]: date,
  [nodesProp]: nodes,
}) =>
  `node.${id}.${date}.${
    nodes ? nodes.map(({ id: idValue }) => idValue).toString() : 'null'
  }.filterAndSortByTime`;

const filterCalculatedEvents = opts => ({ id, grouping, hasTime }) => {
  const groupingValue = ARRAY_HELPERS.arrayIfNot(grouping || opts.grouping).map(
    ({ cacheKey }) => cacheKey,
  );
  const hasTimeValue = hasTime || opts.hasTime;
  return `node.${id}.${groupingValue.toString()}.${hasTimeValue}.filterCalculatedEvents`;
};

const filterByNotMatchingTime = ({ idsProp, nodesProp }) => ({
  [idsProp]: ids,
  [nodesProp]: nodesValue,
}) =>
  `node.${idsProp}:${ids ? ids.toString() : 'null'}.${nodesProp}:${
    nodesValue ? Object.keys(nodesValue).toString() : 'null'
  }.filterByNotMatchingTime`;

const calculatedStartTimeMoment = ({ idProp }) => ({ [idProp]: id }) =>
  `node.${id}.calculatedStartTimeMoment`;

const calculatedEndTimeMoment = ({ idProp }) => ({ [idProp]: id }) =>
  `node.${id}.calculatedEndTimeMoment`;

const calculatedStartEndTimes = ({ idsProp }) => ({ [idsProp]: ids }) =>
  `node.${idsProp}:${ids ? ids.toString() : 'null'}.calculatedStartEndTimes`;

const calculatedStartTimeValues = ({ idsProp, sort }) => ({ [idsProp]: ids }) =>
  `node.${ids ? ids.toString() : 'null'}.${sort}.calculatedStartTimeValues`;

const calculatedTrailParent = ({ idProp }) => ({ [idProp]: id }) =>
  `node.${id}.calculatedTrailParent`;

const cachedCalculatedEvents = ({ idProp }) => ({ [idProp]: id }) =>
  `node.${id}.cachedCalculatedEvents`;

const cachedCalculatedNodeEvents = ({ idProp }) => ({ [idProp]: id }) =>
  `node.${id}.cachedCalculatedNodeEvents`;

const participantsWithSeats = ({ idsProp, secondIdProp }) => ({
  [idsProp]: ids,
  [secondIdProp]: secondId,
}) => `node.${ids ? ids.toString() : 'null'}.${secondId}.participantsWithSeats`;

const participantsWithoutSeats = ({ idsProp, secondIdProp }) => ({
  [idsProp]: ids,
  [secondIdProp]: secondId,
}) =>
  `node.${ids ? ids.toString() : 'null'}.${secondId}.participantsWithoutSeats`;

const templateIdByTrail = ({ idProp }) => ({ [idProp]: id }) =>
  `node.${id}.trails.templateId`;

const confirmedParticipants = ({ idsProp }) => ({ [idsProp]: ids }) =>
  `node.${ids ? ids.toString() : 'null'}.confirmedParticipants`;

const parentNodeIdViaTrail = ({ idProp }) => ({ [idProp]: id }) =>
  `node.${id}.parentNodeIdViaTrail`;

const sortByProp = ({ ids, path, reverse }) => ({
  [ids]: idsPropValue,
  [path]: pathPropValue,
}) => {
  const idsValue = SELECTOR_HELPERS.propOrValueArray(ids, idsPropValue);
  const pathValue = SELECTOR_HELPERS.propOrValueArray(path, pathPropValue);
  return `node.${ids}:${idsValue.toString()}.${path}:${pathValue.toString()}.${reverse}.filterByTypes`;
};

export const NODE_STORE_CACHE_KEYS = {
  checklists,
  allChecklists,
  allStatuses,
  remainingCheckitemsCount,
  openClosedCount,
  cachedChildren,
  cachedIndex,
  eventDataIds,
  addEventDataIds,
  filterByTypes,
  filterByStatuses,
  filterAndSortByTime,
  filterCalculatedEvents,
  filterByNotMatchingTime,
  calculatedStartTimeMoment,
  calculatedEndTimeMoment,
  calculatedStartEndTimes,
  calculatedStartTimeValues,
  calculatedTrailParent,
  cachedCalculatedEvents,
  cachedCalculatedNodeEvents,
  sortByProp,
  participantsWithoutSeats,
  participantsWithSeats,
  confirmedParticipants,
  parentNodeIdViaTrail,
  templateIdByTrail,
};
