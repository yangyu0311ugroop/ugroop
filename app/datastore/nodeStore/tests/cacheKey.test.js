import { NODE_STORE_CACHE_KEYS } from '../cacheKey';

describe('NODE_STORE_CACHE_KEYS', () => {
  describe('remainingCheckitemsCount()', () => {
    it('should return remainingCheckitemsCount cacheKey', () => {
      expect(
        NODE_STORE_CACHE_KEYS.remainingCheckitemsCount({}),
      ).toMatchSnapshot();
      expect(
        NODE_STORE_CACHE_KEYS.remainingCheckitemsCount({ checklists: 123 }),
      ).toMatchSnapshot();
      expect(
        NODE_STORE_CACHE_KEYS.remainingCheckitemsCount({ statuses: 123 }),
      ).toMatchSnapshot();
      expect(
        NODE_STORE_CACHE_KEYS.remainingCheckitemsCount({ type: 'checklist' }),
      ).toMatchSnapshot();
      expect(
        NODE_STORE_CACHE_KEYS.remainingCheckitemsCount({
          checklists: [222],
          statuses: [123],
          type: 'checklist',
        }),
      ).toMatchSnapshot();
      expect(
        NODE_STORE_CACHE_KEYS.remainingCheckitemsCount({
          checklists: [222],
          statuses: [123],
        }),
      ).toMatchSnapshot();
    });
  });

  describe('openClosedCount()', () => {
    it('should return openClosedCount cacheKey', () => {
      expect(
        NODE_STORE_CACHE_KEYS.openClosedCount({ id: [222] }),
      ).toMatchSnapshot();
      expect(NODE_STORE_CACHE_KEYS.openClosedCount({})).toBe('');
      expect(NODE_STORE_CACHE_KEYS.openClosedCount({ id: 112 })).toBe('');
      expect(NODE_STORE_CACHE_KEYS.openClosedCount({ id: false })).toBe('');
      expect(NODE_STORE_CACHE_KEYS.openClosedCount({ id: 'some id' })).toBe('');
    });
  });

  describe('checklists()', () => {
    it('should return checklists cacheKey', () => {
      expect(NODE_STORE_CACHE_KEYS.checklists({ id: 222 })).toMatchSnapshot();
    });
  });

  describe('allChecklists()', () => {
    it('should return allChecklists cacheKey', () => {
      expect(NODE_STORE_CACHE_KEYS.allChecklists({})).toBe(undefined);
      expect(
        NODE_STORE_CACHE_KEYS.allChecklists({ id: 222 }),
      ).toMatchSnapshot();
      expect(
        NODE_STORE_CACHE_KEYS.allChecklists({ id: [1, 2] }),
      ).toMatchSnapshot();
    });
  });

  describe('allStatuses()', () => {
    it('should return allStatuses cacheKey', () => {
      expect(NODE_STORE_CACHE_KEYS.allStatuses({})).toBe(undefined);
      expect(NODE_STORE_CACHE_KEYS.allStatuses({ id: 222 })).toMatchSnapshot();
      expect(
        NODE_STORE_CACHE_KEYS.allStatuses({ id: [1, 2] }),
      ).toMatchSnapshot();
    });
  });

  describe('cachedChildren()', () => {
    it('returns correct key', () => {
      const id = 1;
      const idProp = 'id';
      const props = { [idProp]: id };
      const propNames = { idProp };
      expect(
        NODE_STORE_CACHE_KEYS.cachedChildren(propNames)(props),
      ).toMatchSnapshot();
    });
  });

  describe('index()', () => {
    it('returns correct key', () => {
      const id = 1;
      const idProp = 'id';
      const parentNodeId = 2;
      const parentNodeIdProp = 'parentNodeIdProp';
      const props = {
        [idProp]: id,
        [parentNodeIdProp]: parentNodeId,
      };
      const propNames = { idProp, parentNodeIdProp };
      expect(
        NODE_STORE_CACHE_KEYS.cachedIndex(propNames)(props),
      ).toMatchSnapshot();
    });
  });

  describe('eventDataIds()', () => {
    it('returns correct key', () => {
      const ids = [1, 2];
      const idsProp = 'ids';
      expect(
        NODE_STORE_CACHE_KEYS.eventDataIds({ idsProp })({ [idsProp]: ids }),
      ).toMatchSnapshot();
    });

    it('returns correct key with no props', () => {
      expect(NODE_STORE_CACHE_KEYS.eventDataIds({})({})).toMatchSnapshot();
    });
  });

  describe('addEventDataIds()', () => {
    it('returns correct key', () => {
      const ids = [1, 2];
      const nodesProp = 'ids';
      expect(
        NODE_STORE_CACHE_KEYS.addEventDataIds({ nodesProp })({
          [nodesProp]: ids,
        }),
      ).toMatchSnapshot();
    });

    it('returns correct key with no props', () => {
      expect(NODE_STORE_CACHE_KEYS.addEventDataIds({})({})).toMatchSnapshot();
    });
  });

  describe('filterByTypes()', () => {
    it('returns correct key', () => {
      const idsPropValue = [1, 2];
      const ids = 'ids';
      const typesPropValue = ['a', 'b'];
      const types = 'types';
      const props = { [ids]: idsPropValue, [types]: typesPropValue };
      expect(
        NODE_STORE_CACHE_KEYS.filterByTypes({ ids, types })(props),
      ).toMatchSnapshot();
    });
  });

  describe('filterByStatuses()', () => {
    it('returns correct key', () => {
      const idsPropValue = [1, 2];
      const ids = 'ids';
      const statusesPropValue = ['a', 'b'];
      const statuses = 'statuses';
      const props = { [ids]: idsPropValue, [statuses]: statusesPropValue };
      expect(
        NODE_STORE_CACHE_KEYS.filterByStatuses({ ids, statuses })(props),
      ).toMatchSnapshot();
    });
  });

  describe('filterAndSortByTime()', () => {
    it('returns correct key', () => {
      expect(
        NODE_STORE_CACHE_KEYS.filterAndSortByTime({
          idProp: 'idProp',
          dateProp: 'dateProp',
          nodesProp: 'nodesProp',
        })({
          idProp: 'idProp',
          dateProp: 'dateProp',
          nodesProp: [1],
        }),
      ).toMatchSnapshot();
    });

    it('returns null', () => {
      expect(
        NODE_STORE_CACHE_KEYS.filterAndSortByTime({
          idProp: 'idProp',
          dateProp: 'dateProp',
          nodesProp: 'nodesProp',
        })({
          idProp: 'idProp',
          dateProp: 'dateProp',
        }),
      ).toMatchSnapshot();
    });
  });

  describe('filterCalculatedEvents()', () => {
    it('returns correct key', () => {
      expect(
        NODE_STORE_CACHE_KEYS.filterCalculatedEvents({
          grouping: [{ cacheKey: '1' }],
          hasTime: true,
        })({ id: 333 }),
      ).toMatchSnapshot();
    });

    it('returns null', () => {
      expect(
        NODE_STORE_CACHE_KEYS.filterCalculatedEvents({
          hasTime: true,
        })({ id: 333 }),
      ).toMatchSnapshot();
    });
  });

  describe('filterByNotMatchingTime()', () => {
    it('returns correct key', () => {
      expect(
        NODE_STORE_CACHE_KEYS.filterByNotMatchingTime({
          idsProp: 'idsProp',
          nodesProp: 'nodesProp',
        })({
          idsProp: [1],
          nodesProp: [2],
        }),
      ).toMatchSnapshot();
    });

    it('returns null', () => {
      expect(
        NODE_STORE_CACHE_KEYS.filterByNotMatchingTime({
          idsProp: 'idsProp',
          nodesProp: 'nodesProp',
        })({}),
      ).toMatchSnapshot();
    });
  });

  describe('calculatedStartTimeMoment()', () => {
    it('returns correct key', () => {
      const id = 1;
      const idProp = 'ids';
      expect(
        NODE_STORE_CACHE_KEYS.calculatedStartTimeMoment({ idProp })({
          [idProp]: id,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('calculatedEndTimeMoment()', () => {
    it('returns correct key', () => {
      const id = 1;
      const idProp = 'ids';
      expect(
        NODE_STORE_CACHE_KEYS.calculatedEndTimeMoment({ idProp })({
          [idProp]: id,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('calculatedStartEndTimes()', () => {
    it('returns correct key', () => {
      const ids = [1, 2];
      const idsProp = 'ids';
      expect(
        NODE_STORE_CACHE_KEYS.calculatedStartEndTimes({ idsProp })({
          [idsProp]: ids,
        }),
      ).toMatchSnapshot();
    });

    it('returns correct key with no props', () => {
      expect(
        NODE_STORE_CACHE_KEYS.calculatedStartEndTimes({})({}),
      ).toMatchSnapshot();
    });
  });

  describe('calculatedStartTimeValues()', () => {
    it('returns correct key', () => {
      const ids = [1, 2];
      const idsProp = 'ids';
      const sort = 'sort';
      expect(
        NODE_STORE_CACHE_KEYS.calculatedStartTimeValues({ idsProp, sort })({
          [idsProp]: ids,
        }),
      ).toMatchSnapshot();
    });

    it('returns correct key with no props', () => {
      expect(
        NODE_STORE_CACHE_KEYS.calculatedStartTimeValues({})({}),
      ).toMatchSnapshot();
    });
  });

  describe('calculatedTrailParent()', () => {
    it('returns correct key', () => {
      const id = 1;
      const idProp = 'ids';
      expect(
        NODE_STORE_CACHE_KEYS.calculatedTrailParent({ idProp })({
          [idProp]: id,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('cachedCalculatedEvents()', () => {
    it('returns correct key', () => {
      const id = 1;
      const idProp = 'ids';
      expect(
        NODE_STORE_CACHE_KEYS.cachedCalculatedEvents({ idProp })({
          [idProp]: id,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('sortByProp()', () => {
    it('returns correct key', () => {
      const idsPropValue = [1, 2];
      const ids = 'ids';
      const pathPropValue = ['a', 'b'];
      const path = 'path';
      const props = { [ids]: idsPropValue, [path]: pathPropValue };
      expect(
        NODE_STORE_CACHE_KEYS.sortByProp({ ids, path, reverse: true })(props),
      ).toMatchSnapshot();
    });
  });

  describe('cachedCalculatedEvents', () => {
    it('should return the correct cache key', () => {
      const idProp = 'id';
      expect(
        NODE_STORE_CACHE_KEYS.cachedCalculatedEvents({ idProp })({
          [idProp]: 1,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('calculatedEndTimeMoment', () => {
    it('should return the correct cache key', () => {
      const idProp = 'id';
      expect(
        NODE_STORE_CACHE_KEYS.calculatedEndTimeMoment({ idProp })({
          [idProp]: 1,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('calculatedStartTimeMoment', () => {
    it('should return the correct cache key', () => {
      const idProp = 'id';
      expect(
        NODE_STORE_CACHE_KEYS.calculatedStartTimeMoment({ idProp })({
          [idProp]: 1,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('calculatedStartEndTimes', () => {
    it('should return the correct cache key', () => {
      const idsProp = 'id';
      expect(
        NODE_STORE_CACHE_KEYS.calculatedStartEndTimes({ idsProp })({
          [idsProp]: [1, 2],
        }),
      ).toMatchSnapshot();
    });
  });

  describe('calculatedStartTimeValues', () => {
    it('should return the correct cache key', () => {
      const idsProp = 'id';
      expect(
        NODE_STORE_CACHE_KEYS.calculatedStartTimeValues({ idsProp })({
          [idsProp]: [1, 2],
        }),
      ).toMatchSnapshot();
    });
  });

  describe('participantWithSeats', () => {
    it('should return the correct cache key', () => {
      const idsProp = 'id';
      const secondIdProp = 'secondProp';
      expect(
        NODE_STORE_CACHE_KEYS.participantsWithSeats({ idsProp, secondIdProp })({
          [idsProp]: [1, 2],
          [secondIdProp]: 2,
        }),
      ).toMatchSnapshot();
    });

    it('should return the correct cache key even if idsProp is null', () => {
      const idsProp = 'id';
      const secondIdProp = 'secondProp';
      expect(
        NODE_STORE_CACHE_KEYS.participantsWithSeats({ idsProp, secondIdProp })({
          [idsProp]: null,
          [secondIdProp]: 2,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('participantsWithoutSeats', () => {
    it('should return the correct cache key', () => {
      const idsProp = 'id';
      const secondIdProp = 'secondProp';
      expect(
        NODE_STORE_CACHE_KEYS.participantsWithoutSeats({
          idsProp,
          secondIdProp,
        })({
          [idsProp]: [1, 2],
          [secondIdProp]: 2,
        }),
      ).toMatchSnapshot();
    });

    it('should return the correct cache key event if idsProp is null', () => {
      const idsProp = 'id';
      const secondIdProp = 'secondProp';
      expect(
        NODE_STORE_CACHE_KEYS.participantsWithoutSeats({
          idsProp,
          secondIdProp,
        })({
          [idsProp]: null,
          [secondIdProp]: 2,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('confirmedParticipants', () => {
    it('should return the correct cache key', () => {
      const idsProp = 'id';
      expect(
        NODE_STORE_CACHE_KEYS.confirmedParticipants({ idsProp })({
          [idsProp]: [1, 2],
        }),
      ).toMatchSnapshot();
    });

    it('should return the correct cache key even if idsProp is null', () => {
      const idsProp = 'id';
      expect(
        NODE_STORE_CACHE_KEYS.confirmedParticipants({ idsProp })({
          [idsProp]: null,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('templateIdByTrail', () => {
    it('should return the correct cache key', () => {
      const idProp = 'id';
      expect(
        NODE_STORE_CACHE_KEYS.templateIdByTrail({ idProp })({
          [idProp]: 1,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('parentNodeIdViaTrail', () => {
    it('should return the correct cache key', () => {
      const idProp = 'id';
      expect(
        NODE_STORE_CACHE_KEYS.parentNodeIdViaTrail({ idProp })({
          [idProp]: 1,
        }),
      ).toMatchSnapshot();
    });
  });
  describe('cachedCalculatedNodeEvents', () => {
    it('should return the correct cache key', () => {
      const idProp = 'id';
      expect(
        NODE_STORE_CACHE_KEYS.cachedCalculatedNodeEvents({ idProp })({
          [idProp]: 1,
        }),
      ).toMatchSnapshot();
    });
  });
});
