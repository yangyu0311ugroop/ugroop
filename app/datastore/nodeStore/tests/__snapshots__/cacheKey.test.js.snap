// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NODE_STORE_CACHE_KEYS addEventDataIds() returns correct key 1`] = `"node.ids:,.addEventDataIds"`;

exports[`NODE_STORE_CACHE_KEYS addEventDataIds() returns correct key with no props 1`] = `"node.undefined:null.addEventDataIds"`;

exports[`NODE_STORE_CACHE_KEYS allChecklists() should return allChecklists cacheKey 1`] = `"nodes.222.checklists"`;

exports[`NODE_STORE_CACHE_KEYS allChecklists() should return allChecklists cacheKey 2`] = `"nodes.1,2.checklists"`;

exports[`NODE_STORE_CACHE_KEYS allStatuses() should return allStatuses cacheKey 1`] = `"nodes.222.statuses"`;

exports[`NODE_STORE_CACHE_KEYS allStatuses() should return allStatuses cacheKey 2`] = `"nodes.1,2.statuses"`;

exports[`NODE_STORE_CACHE_KEYS cachedCalculatedEvents should return the correct cache key 1`] = `"node.1.cachedCalculatedEvents"`;

exports[`NODE_STORE_CACHE_KEYS cachedCalculatedEvents() returns correct key 1`] = `"node.1.cachedCalculatedEvents"`;

exports[`NODE_STORE_CACHE_KEYS cachedCalculatedNodeEvents should return the correct cache key 1`] = `"node.1.cachedCalculatedNodeEvents"`;

exports[`NODE_STORE_CACHE_KEYS cachedChildren() returns correct key 1`] = `"node.1.cachedChildren"`;

exports[`NODE_STORE_CACHE_KEYS calculatedEndTimeMoment should return the correct cache key 1`] = `"node.1.calculatedEndTimeMoment"`;

exports[`NODE_STORE_CACHE_KEYS calculatedEndTimeMoment() returns correct key 1`] = `"node.1.calculatedEndTimeMoment"`;

exports[`NODE_STORE_CACHE_KEYS calculatedStartEndTimes should return the correct cache key 1`] = `"node.id:1,2.calculatedStartEndTimes"`;

exports[`NODE_STORE_CACHE_KEYS calculatedStartEndTimes() returns correct key 1`] = `"node.ids:1,2.calculatedStartEndTimes"`;

exports[`NODE_STORE_CACHE_KEYS calculatedStartEndTimes() returns correct key with no props 1`] = `"node.undefined:null.calculatedStartEndTimes"`;

exports[`NODE_STORE_CACHE_KEYS calculatedStartTimeMoment should return the correct cache key 1`] = `"node.1.calculatedStartTimeMoment"`;

exports[`NODE_STORE_CACHE_KEYS calculatedStartTimeMoment() returns correct key 1`] = `"node.1.calculatedStartTimeMoment"`;

exports[`NODE_STORE_CACHE_KEYS calculatedStartTimeValues should return the correct cache key 1`] = `"node.1,2.undefined.calculatedStartTimeValues"`;

exports[`NODE_STORE_CACHE_KEYS calculatedStartTimeValues() returns correct key 1`] = `"node.1,2.sort.calculatedStartTimeValues"`;

exports[`NODE_STORE_CACHE_KEYS calculatedStartTimeValues() returns correct key with no props 1`] = `"node.null.undefined.calculatedStartTimeValues"`;

exports[`NODE_STORE_CACHE_KEYS calculatedTrailParent() returns correct key 1`] = `"node.1.calculatedTrailParent"`;

exports[`NODE_STORE_CACHE_KEYS checklists() should return checklists cacheKey 1`] = `"node.222.checklists"`;

exports[`NODE_STORE_CACHE_KEYS confirmedParticipants should return the correct cache key 1`] = `"node.1,2.confirmedParticipants"`;

exports[`NODE_STORE_CACHE_KEYS confirmedParticipants should return the correct cache key even if idsProp is null 1`] = `"node.null.confirmedParticipants"`;

exports[`NODE_STORE_CACHE_KEYS eventDataIds() returns correct key 1`] = `"node.ids:1,2.eventDataIds"`;

exports[`NODE_STORE_CACHE_KEYS eventDataIds() returns correct key with no props 1`] = `"node.undefined:null.eventDataIds"`;

exports[`NODE_STORE_CACHE_KEYS filterAndSortByTime() returns correct key 1`] = `"node.idProp.dateProp..filterAndSortByTime"`;

exports[`NODE_STORE_CACHE_KEYS filterAndSortByTime() returns null 1`] = `"node.idProp.dateProp.null.filterAndSortByTime"`;

exports[`NODE_STORE_CACHE_KEYS filterByNotMatchingTime() returns correct key 1`] = `"node.idsProp:1.nodesProp:0.filterByNotMatchingTime"`;

exports[`NODE_STORE_CACHE_KEYS filterByNotMatchingTime() returns null 1`] = `"node.idsProp:null.nodesProp:null.filterByNotMatchingTime"`;

exports[`NODE_STORE_CACHE_KEYS filterByStatuses() returns correct key 1`] = `"node.ids:1,2.statuses:a,b.filterByStatuses"`;

exports[`NODE_STORE_CACHE_KEYS filterByTypes() returns correct key 1`] = `"node.ids:1,2.types:a,b.filterByTypes"`;

exports[`NODE_STORE_CACHE_KEYS filterCalculatedEvents() returns correct key 1`] = `"node.333.1.true.filterCalculatedEvents"`;

exports[`NODE_STORE_CACHE_KEYS filterCalculatedEvents() returns null 1`] = `"node.333..true.filterCalculatedEvents"`;

exports[`NODE_STORE_CACHE_KEYS index() returns correct key 1`] = `"node.parentNodeIdProp:2.id:1.index"`;

exports[`NODE_STORE_CACHE_KEYS openClosedCount() should return openClosedCount cacheKey 1`] = `"nodes.222.statuses.open_closed"`;

exports[`NODE_STORE_CACHE_KEYS parentNodeIdViaTrail should return the correct cache key 1`] = `"node.1.parentNodeIdViaTrail"`;

exports[`NODE_STORE_CACHE_KEYS participantWithSeats should return the correct cache key 1`] = `"node.1,2.2.participantsWithSeats"`;

exports[`NODE_STORE_CACHE_KEYS participantWithSeats should return the correct cache key even if idsProp is null 1`] = `"node.null.2.participantsWithSeats"`;

exports[`NODE_STORE_CACHE_KEYS participantsWithoutSeats should return the correct cache key 1`] = `"node.1,2.2.participantsWithoutSeats"`;

exports[`NODE_STORE_CACHE_KEYS participantsWithoutSeats should return the correct cache key event if idsProp is null 1`] = `"node.null.2.participantsWithoutSeats"`;

exports[`NODE_STORE_CACHE_KEYS remainingCheckitemsCount() should return remainingCheckitemsCount cacheKey 1`] = `undefined`;

exports[`NODE_STORE_CACHE_KEYS remainingCheckitemsCount() should return remainingCheckitemsCount cacheKey 2`] = `undefined`;

exports[`NODE_STORE_CACHE_KEYS remainingCheckitemsCount() should return remainingCheckitemsCount cacheKey 3`] = `undefined`;

exports[`NODE_STORE_CACHE_KEYS remainingCheckitemsCount() should return remainingCheckitemsCount cacheKey 4`] = `undefined`;

exports[`NODE_STORE_CACHE_KEYS remainingCheckitemsCount() should return remainingCheckitemsCount cacheKey 5`] = `"nodes.222.123.checklist.outstanding_completed.count"`;

exports[`NODE_STORE_CACHE_KEYS remainingCheckitemsCount() should return remainingCheckitemsCount cacheKey 6`] = `"nodes.222.123..outstanding_completed.count"`;

exports[`NODE_STORE_CACHE_KEYS sortByProp() returns correct key 1`] = `"node.ids:1,2.path:a,b.true.filterByTypes"`;

exports[`NODE_STORE_CACHE_KEYS templateIdByTrail should return the correct cache key 1`] = `"node.1.trails.templateId"`;
